mission_control.lua

```diff
@@ -273,38 +273,36 @@ function MissionControl:focusSpace(space_id, window)
         return
     end
 
-    local do_space_focus = coroutine.wrap(function()
+    if Spaces.focusedSpace() ~= space_id then
+        Spaces.gotoSpace(space_id)
+    end
+
+    local do_window_focus = coroutine.wrap(function()
         if window then
             local function check_focus(win, n)
                 local focused = true
                 for i = 1, n do -- ensure that window focus does not change
                     focused = focused and (Window.focusedWindow() == win)
                     if not focused then return false end
-                    coroutine.yield(false) -- not done
```

Comment:
This triple window focus check with a delay from coroutine.yield is to prevent a false positive where the desired window is initially focused, but then MacOS switches the focus to another window.

This seems to happen when the Mission Control "When switching to an application, switch to a Space with open windows fo the application" setting is enabled.

Please leave this line in so we check that the target window has remained focused for 3x the Window.animationDuration time.